monitor/data_loader.py

```diff
@@ -0,0 +1,279 @@
+import json
+import os
+import time
+
+from elasticsearch import Elasticsearch, helpers
+from elasticsearch.client import IndicesClient
+from elasticsearch.helpers.errors import BulkIndexError
+
+from cli import teos_cli 
+from common.logger import Logger
+
+LOG_PREFIX = "System Monitor"
+logger = Logger(actor="Searcher", log_name_prefix=LOG_PREFIX)
+
+
+class DataLoader:
+    """ 
+    The :class:`DataLoader` is in charge of the monitor's Elasticsearch functionality for loading and searching through data.
+    
+    Args:
+        es_host (:obj:`str`): The host Elasticsearch is listening on.
+        es_port (:obj:`int`): The port Elasticsearch is listening on.
+        api_host (:obj:`str`): The host Teos is listening on.
+        api_port (:obj:`int`): The port Teos is listening on.
+        log_file (:obj:`int`): The path to the log file ES will pull data from. 
+        cloud_id (:obj:`str`): Elasticsearch cloud id, if Elasticsearch Cloud is being used.
+        auth_user (:obj:`str`): Elasticsearch Cloud username, if Elasticsearch Cloud is being used.
+        auth_pw (:obj:`str`): Elasticsearch Cloud password, if Elasticsearch Cloud is being used.
+
+    Attributes:
+        es_host (:obj:`str`): The host Elasticsearch is running on.
+        es_port (:obj:`int`): The port Elasticsearch is runnning on.
+        cloud_id (:obj:`str`): Elasticsearch cloud id, if Elasticsearch Cloud is being used.
+        auth_user (:obj:`str`): Elasticsearch Cloud username, if Elasticsearch Cloud is being used.
+        auth_pw (:obj:`str`): Elasticsearch Cloud password, if Elasticsearch Cloud is being used.
+        es (:obj:`Elasticsearch <elasticsearch.Elasticsearch>`): The Elasticsearch client for searching for data to be visualized.
+        index_client (:obj:`IndicesClient <elasticsearch.client.IndiciesClient>`): The index client where log data is stored.
+        log_path (:obj:`str`): The path to the log file where log file will be pulled from and analyzed by ES.
+        api_host (:obj:`str`): The host Teos is listening on.
+        api_port (:obj:`int`): The port Teos is listening on.
+    
+    """
+
+    def __init__(self, es_host, es_port, api_host, api_port, log_file, cloud_id=None, auth_user=None, auth_pw=None):
+         self.es_host = es_host
+         self.es_port = es_port
+         self.es_cloud_id = cloud_id
+         self.es_auth_user = auth_user
+         self.es_auth_pw = auth_pw
+         if self.es_host is not "":
+             self.es = Elasticsearch([
+                 {'host': self.es_host, 'port': self.es_port}    
+             ])     
+         elif cloud_id is not "":
```

Comment:
This is missing a default else (in case no host not cloud_id is provided). This won't be necessary as long as the config is properly checked before feeding info to the function so it fails early.